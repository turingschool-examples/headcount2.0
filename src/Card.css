.above {
  color: green;
}

.below {
  color: red;
}

.card {
  border: solid 2px black;
  color: black;
}

ul {
list-style: none;
margin: 0;
padding-left: 30px;
padding-bottom: 10px
}

li {
  text-orientation: left;
}

span {
  display: flex;
  justify-content: space-between;
}

h5 {
  padding: 10px;
  margin: 5px;
}

.select-btn {
  padding: 2px;
  border-radius: 25%;
  margin: 5px;
  font-size: 0.25rem;
}