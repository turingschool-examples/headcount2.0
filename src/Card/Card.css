.card {
    box-shadow: 5px 5px 8px;
    width: 185px;
    height: 233px;
    border: 1px #000000 solid;
    margin: 7px;
}

ul {
  list-style-type: none;
}

li {
  font-size: 12px;
}

h3 {
  text-align: center;
  font-size: 15px;
  padding: 0 7px;
}