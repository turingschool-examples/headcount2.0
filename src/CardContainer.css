.cardContainer-grid {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(255px, 1fr));
  grid-gap: 15px;
  padding: 10px;
  margin: 10px;
  border-radius: 25px;
  border: cornflowerblue;
  align-items: center;
}

.cardContainerHeading {
  color: cornflowerblue; 
  text-align: center;
}