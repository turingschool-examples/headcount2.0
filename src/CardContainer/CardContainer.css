.card-container {
  display: flexbox;
  flex-wrap: wrap;
  border: black 1px solid;
  display: flex;
  flex-wrap: wrap;
  width: 84%;
  height: 500px;
  margin: 5px auto;
  justify-content: center;
  overflow-y: scroll;
}