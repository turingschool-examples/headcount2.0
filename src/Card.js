import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Card.css'

const Card = ( {location, stats} ) => {
    const displayStats = Object.keys(stats).map(stat => {
      return ( <li className={`listedStats ${stats[stat] < 0.5 ? 'below' : 'above'}`}> {`${stat}: ${stats[stat] }`}</li>
        )
    })
    
    return(
      <div className='card'>
        <span>
          <h5>{ location }</h5>
          <button className='select-btn'>Select</button>
          </span>
        <ul>
          {displayStats} 
        </ul>
      </div>
    )
  }

Card.propTypes = {
  location: PropTypes.string.isRequired,
  stats: PropTypes.object.isRequired
}

export default Card;