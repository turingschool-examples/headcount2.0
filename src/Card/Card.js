import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Card.css';

class Card extends Component {
  constructor(props) {
    super(props);

    this.state = {
      clicked: this.props.clickedState || false
    };
  }

  handleCardClick = () => {
    
    if (this.props.comparedCards.length === 2  && 
      !this.props.comparedCards.find(district => {
        return district.district.location === this.props.district.location;
      })) {
      return;
    }
    this.setState({ clicked: !this.state.clicked });
    this.props.updateCompareState(this.state.clicked, this.props);
  }

  render() {
    let districtKeys = Object.keys(this.props.district.stats);
    const listItems = districtKeys.map(( stat, index ) => {
      return (
        <li 
          key={ index }
          className={ this.props.district.stats[stat] > 0.5 ? 'greater-than' : 'less-than'}
        > { `${stat}: ${this.props.district.stats[stat]}` } </li>
      );
    });
  
    return (
      <div 
        className={this.state.clicked ? 'card clicked' : 'card'} 
        onClick={ this.handleCardClick }
      > 
        <h3>
          { this.props.district.location }
        </h3>
        <ul>
          { listItems }
        </ul>
      </div>
    );
  }
}

Card.propTypes = {
  district: PropTypes.string.isRequired,
  comparedCards: PropTypes.arrayOf(PropTypes.object),
  updateCompareState: PropTypes.func.isRequired,
  clickedState: PropTypes.bool
};



export default Card;