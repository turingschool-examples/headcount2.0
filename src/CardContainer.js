import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import './CardContainer.css';


const CardContainer = ( {displayData, dataSet} ) => {

  
  const cards = displayData.map(district => {
    return <Card 
            key={`card ${district.location}`}
            {...district} />
  })

  return(
    <div>
      <h3 className='cardContainerHeading'>{dataSet} data for each district is listed below when nothing is entered in the search field, otherwise, only districts that match the search criteria are listed.</h3>
      <section className='cardContainer-grid'>
        { cards }
      </section>
    </div>
  )
}

CardContainer.propTypes = {
  displayData: PropTypes.array.isRequired
}

export default CardContainer;