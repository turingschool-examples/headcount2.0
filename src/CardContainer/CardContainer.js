import React from 'react';
import Card from '../Card/Card.js';
import PropTypes from 'prop-types';
import './CardContainer.css'

const CardContainer = ({districtsData}) => {
  // console.log(districtsData);
  
  const districtCards = Object.keys(districtsData).map((key, index) => {
    return (
      <Card key={index} district={districtsData[key]} />
    )
  });

  return(
    <div className="card-container">
      {districtCards}
    </div>
  )
}

CardContainer.propTypes = {
  districtsData: PropTypes.any.isRequired
}

export default CardContainer;

      