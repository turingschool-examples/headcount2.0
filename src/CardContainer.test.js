import React from 'react';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';
import Card from './Card';

describe('CardContainer', () => {
  it('matches the snapshot', () => {
    let mockDataSet = 'College';
    let mockDisplayData = 
    [{location: 'NorthCarolina', 
    stats: {1999: 0.007, 2099: 1.999 }}, 
    {location: 'VIRGINIA', 
    stats: {1909: 0.107, 3999: 5.999 }}];
    let wrapper = shallow(<CardContainer displayData={mockDisplayData}
      dataSet={mockDataSet}/>);

    expect(wrapper).toMatchSnapshot()
  });

  it('renders all the cards', () => {
    let mockDataSet = 'College';
    let mockDisplayData = 
    [ {location: 'NorthCarolina', 
    stats: {1999: 0.007, 2099: 1.999 }}, 
    {location: 'VIRGINIA', 
    stats: {1909: 0.107, 3999: 5.999 }}];

    let wrapper = shallow(<CardContainer
        displayData={ mockDisplayData }
        dataSet={mockDataSet} />)
    expect(wrapper.find(Card).length).toEqual(2)
  });

  it.skip('Assigns a unique key to each card', () => {


  });

})