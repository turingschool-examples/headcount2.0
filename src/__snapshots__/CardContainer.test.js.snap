// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    dataSet="College"
    displayData={
        Array [
            Object {
              "location": "NorthCarolina",
              "stats": Object {
                "1999": 0.007,
                "2099": 1.999,
              },
            },
            Object {
              "location": "VIRGINIA",
              "stats": Object {
                "1909": 0.107,
                "3999": 5.999,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="cardContainerHeading"
>
          College
           data for each district is listed below when nothing is entered in the search field, otherwise, only districts that match the search criteria are listed.
</h3>,
        <section
          className="cardContainer-grid"
>
          <Card
                    location="NorthCarolina"
                    stats={
                              Object {
                                        "1999": 0.007,
                                        "2099": 1.999,
                                      }
                    }
          />
          <Card
                    location="VIRGINIA"
                    stats={
                              Object {
                                        "1909": 0.107,
                                        "3999": 5.999,
                                      }
                    }
          />
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "College",
            " data for each district is listed below when nothing is entered in the search field, otherwise, only districts that match the search criteria are listed.",
          ],
          "className": "cardContainerHeading",
        },
        "ref": null,
        "rendered": Array [
          "College",
          " data for each district is listed below when nothing is entered in the search field, otherwise, only districts that match the search criteria are listed.",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              location="NorthCarolina"
              stats={
                            Object {
                                          "1999": 0.007,
                                          "2099": 1.999,
                                        }
              }
/>,
            <Card
              location="VIRGINIA"
              stats={
                            Object {
                                          "1909": 0.107,
                                          "3999": 5.999,
                                        }
              }
/>,
          ],
          "className": "cardContainer-grid",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "card NorthCarolina",
            "nodeType": "function",
            "props": Object {
              "location": "NorthCarolina",
              "stats": Object {
                "1999": 0.007,
                "2099": 1.999,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "card VIRGINIA",
            "nodeType": "function",
            "props": Object {
              "location": "VIRGINIA",
              "stats": Object {
                "1909": 0.107,
                "3999": 5.999,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="cardContainerHeading"
>
            College
             data for each district is listed below when nothing is entered in the search field, otherwise, only districts that match the search criteria are listed.
</h3>,
          <section
            className="cardContainer-grid"
>
            <Card
                        location="NorthCarolina"
                        stats={
                                    Object {
                                                "1999": 0.007,
                                                "2099": 1.999,
                                              }
                        }
            />
            <Card
                        location="VIRGINIA"
                        stats={
                                    Object {
                                                "1909": 0.107,
                                                "3999": 5.999,
                                              }
                        }
            />
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "College",
              " data for each district is listed below when nothing is entered in the search field, otherwise, only districts that match the search criteria are listed.",
            ],
            "className": "cardContainerHeading",
          },
          "ref": null,
          "rendered": Array [
            "College",
            " data for each district is listed below when nothing is entered in the search field, otherwise, only districts that match the search criteria are listed.",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                location="NorthCarolina"
                stats={
                                Object {
                                                "1999": 0.007,
                                                "2099": 1.999,
                                              }
                }
/>,
              <Card
                location="VIRGINIA"
                stats={
                                Object {
                                                "1909": 0.107,
                                                "3999": 5.999,
                                              }
                }
/>,
            ],
            "className": "cardContainer-grid",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "card NorthCarolina",
              "nodeType": "function",
              "props": Object {
                "location": "NorthCarolina",
                "stats": Object {
                  "1999": 0.007,
                  "2099": 1.999,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "card VIRGINIA",
              "nodeType": "function",
              "props": Object {
                "location": "VIRGINIA",
                "stats": Object {
                  "1909": 0.107,
                  "3999": 5.999,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
